### DISCUSSION TITLE
NativeWind v4 style with solito link

### DISCUSSION THREAD

**Discussion by MustafaAlsihati** [nativewind/nativewind#1343] [Q&A]
Hi, I have a question regarding styling Solito text after upgrading to NativeWind v4, before v4 we had `styled()` function which adds className prop and apply styles.

Now since we don't have this function anymore I couldn't apply tailwind styles to Solito text.

v2:
```
export const TextLink = styled<
  ComponentProps<typeof SolitoTextLink> & { style?: TextStyle }
>(function TextLink({ style, textProps, ...props }) {
  return (
    <SolitoTextLink
      textProps={{ ...textProps, style: [style, textProps?.style] }}
      {...props}
    />
  );
}, '!font-custom-medium text-base hover:underline text-link');
```

v4:
```
type TextLinkProps = ComponentProps<typeof SolitoTextLink> & {
  style?: TextStyle;
};
export const TextLink = function TextLink({
  style,
  textProps,
  ...props
}: TextLinkProps) {
  const baseClassName =
    '!font-custom-medium text-base hover:underline text-link';
  return (
    <SolitoTextLink
      {...props}
      textProps={{
        ...textProps,
        style: [style, textProps?.style],
        className: cn(baseClassName, props.className),
      }}
      className={cn(baseClassName, props.className)}
    />
  );
};
```

I tried adding className directly to textProps but still not working, I also tried using both `cssInterop` and `remapProps` but no luck, maybe I didn't use them the correct way?

Any idea how to solve this issue?

**Comment by danstepanov** [comment]
`styled()` is once again available via Nativewind v5: https://www.nativewind.dev/v5/guides/migrate-from-v4#cssinterop--remapprops--replaced-by-styled
